dotcom-rendering/src/web/components/LiveEpic.importable.tsx

```diff
@@ -0,0 +1,288 @@
+	}
+
+	return false;
+}
+
+/**
+ * usePayload
+ *
+ * takes a series of CAPI values, reads consent, the users country, some cookie
+ * values and then constructs a config object with `tracking` and `targeting`
+ * properties
+ *
+ * @returns the payload object required to make the POST request to contributions
+ */
+const usePayload = ({
+	shouldHideReaderRevenue,
+	section,
+	isPaidContent,
+	tags,
+}: {
+	shouldHideReaderRevenue: boolean;
+	section: string;
+	isPaidContent: boolean;
+	tags: TagType[];
+}) => {
+	const consent = useConsent();
+	const countryCode = useCountryCode();
+	const mvtId = getCookie({ name: 'GU_mvt_id', shouldMemoize: true }) || 0;
+	const isSignedIn = !!getCookie({ name: 'GU_U', shouldMemoize: true });
+
+	if (!consent) return;
+	log('dotcom', 'LiveEpic has consent state');
+	if (!countryCode) return;
+	log('dotcom', 'LiveEpic has countryCode');
+
+	return {
+		tracking: {
+			ophanPageId: window.guardian.config.ophan.pageViewId,
+			platformId: 'GUARDIAN_WEB',
+			clientName: 'dcr',
+			referrerUrl: window.location.origin + window.location.pathname,
+		},
+		targeting: {
+			contentType: 'LiveBlog',
+			sectionId: section,
+			shouldHideReaderRevenue: shouldHideReaderRevenue ?? false,
+			isMinuteArticle: true,
+			isPaidContent,
+			tags,
+			showSupportMessaging: !shouldHideSupportMessaging(isSignedIn),
+			isRecurringContributor: isRecurringContributor(isSignedIn),
+			lastOneOffContributionDate:
+				getLastOneOffContributionTimestamp() ?? undefined,
+			mvtId,
+			countryCode,
+			epicViewLog: storage.local.get('gu.contributions.views') || [],
+			weeklyArticleHistory:
+				storage.local.get('gu.history.weeklyArticleCount') || [],
+			hasOptedOutOfArticleCount: !consentGivenForArticleCounts(consent),
+			modulesVersion: 'v3',
+			url: window.location.origin + window.location.pathname,
+		},
+	};
+};
+
+/**
+ * Render
+ *
+ * Dynamically imports and renders a given module
+ *
+ * @param url string - The url of the compoenent
+ * @param name string - The name of the compoenent
+ * @param props object - The props of the compoenent
+ * @returns The resulting react component
+ */
+const Render = ({
+	url,
+	name,
+	props,
+}: {
+	url: string;
+	name: string;
+	props: { [key: string]: unknown };
+}) => {
+	// Using state here to store the Epic component that gets imported allows
+	// us to render it with React (instead of inserting it into the dom manually)
+	const [Epic, setEpic] = useState<React.FC<{ [key: string]: unknown }>>();
+
+	useEffect(() => {
+		setAutomat();
+		window
+			.guardianPolyfilledImport(url)
+			.then((epicModule) => {
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+				setEpic(() => epicModule[name]); // useState requires functions to be wrapped
+			})
+			.catch((err) => {
+				const msg = `Error importing LiveBlog epic: ${err}`;
+				window.guardian.modules.sentry.reportError(
+					new Error(msg),
+					'liveblog-epic',
+				);
+			});
+	}, [url, name]);
```

Comment:
could this be factored into a `useEpic` hook?

Reply:
Great idea!